File: Python/Program/Storage.py
from KomponenInternal import KomponenInternal

class Storage(KomponenInternal):
    def __init__(self, merk="", nama="", harga=0, jenis_slot="", terpasang=False, kapasitas_gb=0, kecepatan_baca=0, kecepatan_tulis=0):
        super().__init__(merk, nama, harga, jenis_slot, terpasang)
        self.kapasitas_gb = kapasitas_gb
        self.kecepatan_baca = kecepatan_baca
        self.kecepatan_tulis = kecepatan_tulis
    
    # Setter
    def set_kapasitas_gb(self, kapasitas_gb):
        self.kapasitas_gb = kapasitas_gb
    
    def set_kecepatan_baca(self, kecepatan_baca):
        self.kecepatan_baca = kecepatan_baca
    
    def set_kecepatan_tulis(self, kecepatan_tulis):
        self.kecepatan_tulis = kecepatan_tulis
    
    # Getter
    def get_kapasitas_gb(self):
        return self.kapasitas_gb
    
    def get_kecepatan_baca(self):
        return self.kecepatan_baca
    
    def get_kecepatan_tulis(self):
        return self.kecepatan_tulis
    
    # Tampilkan Info
    def tampilkan_info(self):
        return f"{super().tampilkan_info()} | Kapasitas: {self.kapasitas_gb} GB | R/W: {self.kecepatan_baca}/{self.kecepatan_tulis} MB/s"

File: Java/Program/Storage.java
public class Storage extends KomponenInternal {
    private int kapasitasGB;
    private int kecepatanBaca;
    private int kecepatanTulis;

    // Constructor
    public Storage() {
        super();
        this.kapasitasGB = 0;
        this.kecepatanBaca = 0;
        this.kecepatanTulis = 0;
    }

    public Storage(String merk, String nama, int harga, String jenisSlot, boolean terpasang,
                   int kapasitasGB, int kecepatanBaca, int kecepatanTulis) {
        super(merk, nama, harga, jenisSlot, terpasang);
        this.kapasitasGB = kapasitasGB;
        this.kecepatanBaca = kecepatanBaca;
        this.kecepatanTulis = kecepatanTulis;
    }

    // Setter methods
    public void setKapasitasGB(int kapasitasGB) { this.kapasitasGB = kapasitasGB; }
    public void setKecepatanBaca(int kecepatanBaca) { this.kecepatanBaca = kecepatanBaca; }
    public void setKecepatanTulis(int kecepatanTulis) { this.kecepatanTulis = kecepatanTulis; }

    // Getter methods
    public int getKapasitasGB() { return kapasitasGB; }
    public int getKecepatanBaca() { return kecepatanBaca; }
    public int getKecepatanTulis() { return kecepatanTulis; }

    // Tampilkan Info
    @Override
    public String tampilkanInfo() {
        return super.tampilkanInfo() + " | Kapasitas: " + kapasitasGB + " GB | R/W: " + kecepatanBaca + "/" + kecepatanTulis + " MB/s";
    }
}